server.js

```diff
@@ -0,0 +1,59 @@
+'use strict';
+
+const net = require('net');
+const EE = require('events');
+const Client = require('./model/client.js');
+const PORT = process.env.PORT || 3000;
+const server = net.createServer();
+const ee = new EE();
+
+const pool = [];
+
+ee.on('@dm', (client, string) => {
+  let nickname = string.split(' ').shift().trim();
+  let message = string.split(' ').splice(1).trim();
+
+  pool.forEach((client) => {
+    if(client.nickname === nickname)
+      client.socket.write(`${client.nickname}: ${message}`);
+  });
+});
+
+ee.on('@all', (client, string) => {
+  pool.forEach((c) => {
+    c.socket.write(`${client.nickname}: ${message}`);
+  });
+});
+
+ee.on('@rename', (client, string) => {
+  client.nickname = string.split(' ').shift().trim();
+});
+
+ee.on('default', (client, string) => {
+  client.socket.write('not a command \n');
+});
+
+server.on('connection', (socket) => {
+  var client = new Client(socket);
+  pool.push(Client);
+
+  socket.on('data', (socket) => {
```

Comment:
you are passing in socket here to this function. This is actually the data part so the better word to use would be data. It is also erring out on the next line when you try to access data.